specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2023-11-01-preview/sqlServerAvailabilityGroups.json

```diff
@@ -0,0 +1,802 @@
+        },
+        "operationalStateDescription": {
+          "type": "string",
+          "description": "Current operational state of the availability group replica"
+        },
+        "recoveryHealthDescription": {
+          "type": "string",
+          "description": "Recovery health of the availability group replica."
+        },
+        "synchronizationHealthDescription": {
+          "type": "string",
+          "description": "Reflects a rollup of the database synchronization state (synchronization_state) of all joined availability databases (also known as replicas) and the availability mode of the replica (synchronous-commit or asynchronous-commit mode). The rollup will reflect the least healthy accumulated state the databases on the replica."
+        },
+        "connectedStateDescription": {
+          "type": "string",
+          "description": "Whether a secondary replica is currently connected to the primary replica."
+        },
+        "lastConnectErrorDescription": {
+          "type": "string",
+          "description": "Text description of the last connection error of the availability group replica."
+        },
+        "lastConnectErrorTimestamp": {
+          "type": "string",
+          "description": "Date and time timestamp indicating when the last connect error occurred.",
+          "format": "date-time"
+        }
+      }
+    },
+    "AvailabilityGroupInfo": {
+      "description": "The specifications of the availability group state",
+      "type": "object",
+      "properties": {
+        "failureConditionLevel": {
+          "type": "integer",
+          "description": "User-defined failure condition level under which an automatic failover must be triggered.",
+          "format": "int32"
+        },
+        "healthCheckTimeout": {
+          "type": "integer",
+          "description": "Wait time (in milliseconds) for the sp_server_diagnostics system stored procedure to return server-health information, before the server instance is assumed to be slow or not responding.",
+          "format": "int32"
+        },
+        "automatedBackupPreferenceDescription": {
+          "type": "string",
+          "description": "Preferred location for performing backups on the availability databases in this availability group."
+        },
+        "version": {
+          "type": "integer",
+          "description": "SQL Server availability group current version.",
+          "format": "int32"
+        },
+        "basicFeatures": {
+          "type": "boolean",
+          "description": "Specifies whether this is a basic availability group."
+        },
+        "dtcSupport": {
+          "type": "boolean",
+          "description": "Specifies whether DTC support has been enabled for this availability group."
+        },
+        "dbFailover": {
+          "type": "boolean",
+          "description": "Specifies whether the availability group supports failover for database health conditions."
+        },
```

Comment:
Booleans are not future proof, and I'd like to gently discourage you from using booleans for any property where you *may want to extend its range of meanings in the future*.

Reply:
For example instead of saying 'basicFeatures' = true/false

Maybe you want to say featureSet = basic.

Reply:
For example instead of saying 'dbFailover = true/false'

Maybe you want to say 'dbFailoverStrategy' = 'Automatic/Manual/None'